test/AutocompleteBadges.js

```diff
@@ -128,4 +128,146 @@ describe('AutocompleteBadges', function() {
 		}, false);
 		assert.doesNotThrow(() => component.dispose());
 	});
+
+
+	it('should NOT select the first badge when backspace is pressed and the input has some text', function(done) {
+		component = new AutocompleteBadges({
+			dataItems: elements
+		});
+
+		component.getInput().value = 'a';
+		dom.triggerEvent(component.getInput(), 'input');
+
+		component.getAutocomplete().getList().once('rendered', function() {
+			component.getAutocomplete().once('select', function () {
+				component.once('rendered', function () {
+					component.getInput().value = 'some text';
+					dom.triggerEvent(component.getInput(), 'keydown', {keyCode : 8});
+					assert.strictEqual(component.getInput(), document.activeElement);
+					done();
+				});
+			});
+			dom.triggerEvent(component.getAutocomplete().element.querySelectorAll('li')[0], 'click');
+		});
+
+	});
+
+	it('should not do anything when left arrow is pressed and there are no badges', function(done) {
+		component = new AutocompleteBadges({
+			dataItems: elements
+		});
+
+		dom.triggerEvent(component.getInput(), 'keydown', {keyCode : 37});
+		assert.strictEqual(0, component.element.querySelectorAll('.autocomplete-badges--list li:focus').length);
+		done();
+
+	});
+
+	it('should navigate badges focus with left and right arrow keys', function(done) {
+		component = new AutocompleteBadges({
+			dataItems: elements
+		});
+
+		component.getInput().value = 'a';
+		dom.triggerEvent(component.getInput(), 'input');
+
+		component.getAutocomplete().getList().once('rendered', function() {
```

Comment:
Thanks for writing these tests! They're covering the features really well 😄 

The only problem is that, due to the initial setup of adding some badges, they're hard to read. Maybe we could add a helper for adding badges, so this can become cleaner. Just having a function in this file for this, something like `addBadgesHelper_(count, callback)` maybe, would already help. With it this test could be just:

```js
it('should navigate badges focus with left and right arrow keys', function(done) {
  component = new AutocompleteBadges({
    dataItems: elements
  });

  addBadgesHelper_(2, () => {
    dom.triggerEvent(component.getInput(), 'keydown', {keyCode : 37});
    let badges = component.element.querySelectorAll('.autocomplete-badges--list li');
    assert.strictEqual(badges[1],document.activeElement);
    dom.triggerEvent(badges[1], 'keydown', {keyCode : 37});
    assert.strictEqual(badges[0],document.activeElement);
    dom.triggerEvent(badges[0], 'keydown', {keyCode : 37});
    assert.strictEqual(badges[0],document.activeElement);
    dom.triggerEvent(badges[0], 'keydown', {keyCode : 39});
    assert.strictEqual(badges[1],document.activeElement);
    dom.triggerEvent(badges[1], 'keydown', {keyCode : 39});
    assert.strictEqual(component.getInput(),document.activeElement);
    done();
  });
});
```